contracts/ERC721A.sol

```diff
@@ -107,74 +120,106 @@ contract ERC721A is IERC721A {
      */
     function balanceOf(address owner) public view override returns (uint256) {
         if (owner == address(0)) revert BalanceQueryForZeroAddress();
-        return uint256(_addressData[owner].balance);
+        return _packedAddressData[owner] & ((1 << 64) - 1);
     }
 
     /**
      * Returns the number of tokens minted by `owner`.
      */
     function _numberMinted(address owner) internal view returns (uint256) {
-        return uint256(_addressData[owner].numberMinted);
+        return (_packedAddressData[owner] >> 64) & ((1 << 64) - 1);
```

Comment:
I think we can make these bit masks constants so the shift operation only needs to be done once. It also gives us some clearer naming

Reply:
The compiler will evaluate the constant expressions on compile time.

Reply:
Even so, I prefer not to have magic numbers in our code if possible

Reply:
Ok. The code will be more verbose though.